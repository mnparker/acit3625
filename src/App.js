import React, { useState, useContext} from "react";
import { Route, Switch } from 'react-router-dom';

import { createGlobalStyle} from "styled-components"
import ThemeContext from './ThemeContext'
import AppTheme from "./Colours";

import Home from './components/Home'
import About from './components/About'
import NavBar from './components/Navbar'
import Error from './components/Error'
import API from './components/API'



function App(props) {
  
  const themeHook = useState('light');
  
  const theme = useContext(ThemeContext)[0];
  const currentTheme = AppTheme[theme];
  
  const GlobalStyles = createGlobalStyle`
  body {
    background: ${currentTheme.body};
  }
  `
  return (

    <ThemeContext.Provider value = {themeHook}>
      <GlobalStyles/>
      <NavBar />
        <Switch>
          <Route path='/' component={Home} exact />
          <Route path='/about' component={About} />
          <Route path='/api' component={API} />
          <Route component={Error} />
        </Switch>
    </ThemeContext.Provider>
   
  );
}

export default App;
