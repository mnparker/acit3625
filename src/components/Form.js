import React, { useState, useContext } from "react";
import AppTheme from "../Colours";
import ThemeContext from "../ThemeContext";

function Form(props) {

    const [name, setName] = useState('');

    const theme = useContext(ThemeContext)[0];
    const currentTheme = AppTheme[theme];

    function handleSubmit(e) {
        e.preventDefault();
        if (name !== '') {
            props.addTask(name)};
        setName("");
    }

    function handleChange(e) {
        setName(e.target.value);
    }

  return (
    <form onSubmit={handleSubmit}>
      <h2 className="label-wrapper">
        <label htmlFor="new-todo-input" className="label__lg">
          What needs to be done?
        </label>
      </h2>
      <input
      style = {{
        backgroundColor: `${currentTheme.backgroundColor}`,
        color: `${currentTheme.textColor}`,
        borderColor: `${currentTheme.borderColor}`
    }}
        type="text"
        id="new-todo-input"
        className="input input__lg"
        name="text"
        autoComplete="off"
        value={name}
        onChange={handleChange}
      />
      <button 
      style = {{
        backgroundColor: `${currentTheme.saveColor}`,
        borderColor: `${currentTheme.borderColor}`
      }}
      type="submit" className="btn btn__primary btn__lg">
        Add
      </button>
    </form>
  );
}

export default Form;