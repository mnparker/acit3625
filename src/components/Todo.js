import React, { useState, useContext } from "react";
import AppTheme from "../Colours";
import ThemeContext from "../ThemeContext";


export default function Todo(props) {
    const [isEditing, setEditing] = useState(false);
    const [newName, setNewName] = useState('');

    const theme = useContext(ThemeContext)[0];
    const currentTheme = AppTheme[theme];

    function handleChange(e) {
        setNewName(e.target.value);
      }


      function handleSubmit(e) {
        e.preventDefault();
        props.editTask(props.id, newName);
        setNewName("");
        setEditing(false);
      }



    const editingTemplate = (
        <form className="stack-small" onSubmit={handleSubmit}>
          <div className="form-group">
            <label className="todo-label" htmlFor={props.id}>
              New name for {props.name}
            </label>
            <input 
            style = {{
              backgroundColor: `${currentTheme.backgroundColor}`,
              color: `${currentTheme.textColor}`,
              borderColor: `${currentTheme.borderColor}`
          }}
            id={props.id} 
            className="todo-text" 
            type="text" 
            value={newName}
            onChange={handleChange}
            />
          </div>
          <div className="btn-group">
            <button 
            style={{
              borderColor: `${currentTheme.borderColor}`
            }}
            
            type="button" className="btn todo-cancel" onClick={() => setEditing(false)}>
              Cancel
              <span className="visually-hidden">renaming {props.name}</span>
            </button>
            <button
            style = {{
              backgroundColor: `${currentTheme.saveColor}`,
              borderColor: `${currentTheme.borderColor}`
            }}
            type="submit" className="btn btn__primary todo-edit">
              Save
              <span className="visually-hidden">new name for {props.name}</span>
            </button>
          </div>
        </form>
      );
      const viewTemplate = (
        <div className="stack-small">
          <div className="c-cb">
              <input
                id={props.id}
                type="checkbox"
                defaultChecked={props.completed}
                onChange={() => props.toggleTaskCompleted(props.id)}
              />
              <label className="todo-label" htmlFor={props.id}>
                {props.name}
              </label>
            </div>
            <div className="btn-group">
              <button style={{
                          color: `${currentTheme.textColor2}`,
                          backgroundColor: `${currentTheme.buttonColor2}`,
                          borderColor: `${currentTheme.borderColor}`
                        }}
              
              type="button" className="btn" onClick={() => setEditing(true)}>
                Edit <span className="visually-hidden">{props.name}</span>
              </button>
              <button style={{
                          backgroundColor: `${currentTheme.buttonColor}`,
                          borderColor: `${currentTheme.borderColor}`
                        }}
                type="button"
                className="btn btn__danger"
                onClick={() => props.deleteTask(props.id)}
              >
                Delete <span className="visually-hidden">{props.name}</span>
              </button>
            </div>
        </div>
      );

    return <li className="todo">{isEditing ? editingTemplate : viewTemplate}
    </li>;

  }