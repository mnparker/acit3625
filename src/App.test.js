import React from "react";
import renderer from 'react-test-renderer';
import App from './App';
import Form from './components/Form'
import FilterButton from './components/FilterButton'
import Todo from './components/Todo'
import { shallow } from 'enzyme';


const data = [
    { id: "todo-0", name: "Eat", completed: true },
    { id: "todo-1", name: "Sleep", completed: false },
    { id: "todo-2", name: "Repeat", completed: false }
  ];



describe('Test Rendering', () => {
  it('renders title of app', () => {
    const wrapper = shallow(<App 
    tasks={data}/>);
    const title =  <h1>TodoMatic</h1>;
    expect(wrapper.contains(title)).toEqual(true);
  });

  it('renders Form correctly', () => {
    const wrapper = shallow(<Form />);
    const label = <h2 className="label-wrapper">
          <label htmlFor="new-todo-input" className="label__lg">
            What needs to be done?
          </label>
        </h2>;
      expect(wrapper.contains(label)).toEqual(true);
    });

    it('should say 3 tasks remaining', () => {
      const wrapper = shallow(<App tasks={data}/>);
      const heading = <h2 id="list-heading">3 tasks remaining</h2>;
      expect(wrapper.contains(heading)).toBe(true);
     
      });
    
});


describe('Snapshots', () => {
  it('renders App component correctly', () => {
      const component = renderer.create(<Form />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });


  it('renders Form correctly', () => {
    const component = renderer.create(<FilterButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});



