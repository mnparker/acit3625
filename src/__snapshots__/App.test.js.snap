// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders App component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2
    className="label-wrapper"
  >
    <label
      className="label__lg"
      htmlFor="new-todo-input"
    >
      What needs to be done?
    </label>
  </h2>
  <input
    autoComplete="off"
    className="input input__lg"
    id="new-todo-input"
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn__primary btn__lg"
    type="submit"
  >
    Add
  </button>
</form>
`;

exports[`Snapshots renders Form correctly 1`] = `
<button
  className="btn toggle-btn"
  onClick={[Function]}
  type="button"
>
  <span
    className="visually-hidden"
  >
    Show 
  </span>
  <span>
     
  </span>
  <span
    className="visually-hidden"
  >
     tasks
  </span>
</button>
`;
